### PR TITLE
[IMPROVEMENT] Add Support for Custom Headers and HTTP Methods in Generic Webhook

### PR DESCRIPTION
**Pull request by Siddhanttimeline**
### Describe your changes:

Fixes https://github.com/open-metadata/OpenMetadata/issues/17879

- Support for custom headers and http methods (POST/PUT) in generic webhook. 

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-spec/src/main/resources/json/schema/entity/events/webhook.json [hunk 1]
```diff
@@ -35,6 +35,21 @@
       "description": "Secret set by the webhook client used for computing HMAC SHA256 signature of webhook payload and sent in `X-OM-Signature` header in POST requests to publish the events.",
       "type": "string"
     },
+    "headers": {
+      "description": "Custom headers to be sent with the webhook request.",
+      "type": "object",
+      "existingJavaType": "java.util.Map<String, String>"
+    },
+    "json": {
```
Comment (mohityadav766) [inline openmetadata-spec/src/main/resources/json/schema/entity/events/webhook.json:43]: any reason to add  json here @Siddhanttimeline ?

Comment (mohityadav766) [reply openmetadata-spec/src/main/resources/json/schema/entity/events/webhook.json:43]: let's remove that

Comment (Siddhanttimeline) [reply openmetadata-spec/src/main/resources/json/schema/entity/events/webhook.json:43]: Noted.

**Comment by Siddhanttimeline** [comment]
@aniketkatkar97 We will require changes in the UI side for this.
1. Field for providing custom headers for webhook.
2. Field for choosing between POST and PUT http method for webhook.

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/changeEvent/generic/GenericPublisher.java [hunk 2]
```diff
@@ -67,79 +69,89 @@ public GenericPublisher(
...
+      if (!CommonUtil.nullOrEmpty(headers)) {
+        headers.forEach(target::header);
+      }
+
+      Webhook.HttpMethod httpOperation = webhook.getHttpMethod();
+      postWebhookMessage(this, target, json, httpOperation);
+    }
+  }
+
+  private void handleException(long attemptTime, ChangeEvent event, Exception ex)
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/changeEvent/generic/GenericPublisher.java:121]: why are there handleException , i see most of the code is same , can you remove the duplicate code

Comment (Siddhanttimeline) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/changeEvent/generic/GenericPublisher.java:121]: Noted.

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**